### PR TITLE
:sparkles: feat(annotations): add Domain Entity annotation

### PR DESCRIPTION
**Pull request by Yugo-Cartron**
# Pull request for Candy-doc

## Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../../CONTRIBUTING.md) to
this repository.

- [x] Make sure you are requesting to **pull a feature/bugfix branch** (right
  side). Don't request your master!
- [x] Make sure you are making a pull request against the **main branch** (left
  side).
- [x] Check the commit's or even all commits' message styles matches our
  requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit
  test.

## Description

Please describe your pull request.

❤️ Thank you!

### CODE REVIEW

**Review by ctacquet — DISMISSED** [review]
Peux-tu expliquer à quoi correspond une domain entity stp ?

File: src/main/java/io/candydoc/ddd/tactical_pattern/DomainEvent.java [hunk 1]
```diff
@@ -11,7 +11,8 @@
 @ToString(callSuper = true)
 public class DomainEvent extends TacticalModel {
 
-  private static final Set<DDDPattern> ALLOWED_DEPENDENCIES = Set.of(TacticalPattern.VALUE_OBJECT);
+  private static final Set<DDDPattern> ALLOWED_DEPENDENCIES =
```
[the remark this answers is not in the document]
Comment (Yugo-Cartron) [reply src/main/java/io/candydoc/ddd/tactical_pattern/DomainEvent.java:14]: Yes you're right. Then we removed DOMAIN_ENTITY as a DOMAIN_COMMAND/DOMAIN_ENTITY allowed dependencies. We'll work on it, on a DomainEntityId PR

File: src/test/java/io/candydoc/ddd/tactical_patterns/domain_entity/DomainEntityRulesTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                    .build())
+            .build();
+  }
+
+  @Test
+  void is_a_domain_entity() {
+    assertThat(entity.dddPattern()).isEqualTo(TacticalPattern.DOMAIN_ENTITY);
+  }
+
+  @Test
```
Comment (ecattez) [inline src/test/java/io/candydoc/ddd/tactical_patterns/domain_entity/DomainEntityRulesTest.java:55]: As we keep adding the new concepts, I think our tests `it_can_depend` and `it_can_not_depend` should be improved to handle a list of all allowed and disallowed concepts. We could do it through Parameterized Test and a source of Tactical / Strategic Patterns. IMO, we should do it on a specific PR to do this only one time for each concept

**Review by ecattez — APPROVED** [review]

File: src/test/java/io/candydoc/ddd/tactical_patterns/domain_entity/DomainEntityRulesTest.java (hunk 2 above)
Comment (ecattez) [reply src/test/java/io/candydoc/ddd/tactical_patterns/domain_entity/DomainEntityRulesTest.java:55]: I see that you started to do something like that in this PR: https://github.com/Candy-Doc/candy-doc-maven-plugin/pull/195#pullrequestreview-1467391975

Comment (Yugo-Cartron) [reply src/test/java/io/candydoc/ddd/tactical_patterns/domain_entity/DomainEntityRulesTest.java:55]: It's a complex one. I mean, it's hard to retrieve the list of allowed_dependencies without using the _ALLOWED_DEPENDENCIES_ field. If we use it anyway, It'll look like the snake is bitting his own tail.

Comment (Yugo-Cartron) [reply src/test/java/io/candydoc/ddd/tactical_patterns/domain_entity/DomainEntityRulesTest.java:55]: Actually, it's doing well with _@repository_, given that just the aggregate is allowed as dependency.

**Review by ecattez — APPROVED** [review]

**Pull request merged**